core/arch/arm/mm/core_mmu.c

```diff
@@ -480,3 +492,222 @@ unsigned int cache_maintenance_l2(int op, paddr_t pa, size_t len)
 	return ret;
 }
 #endif /*CFG_PL310*/
+
+static bool arm_va2pa_helper(void *va, paddr_t *pa)
+{
+	uint32_t exceptions = thread_mask_exceptions(THREAD_EXCP_ALL);
+	bool ret = false;
+
+#ifdef ARM32
+#ifdef CFG_WITH_LPAE
+	uint64_t par;
+
+	write_ats1cpw((uint32_t)va);
+	isb();
+	par = read_par64();
+	if (par & PAR_F)
+		goto out;
+	*pa = (par & (PAR64_PA_MASK << PAR_PA_SHIFT)) |
```

Comment:
It looks to me that the expression `*pa = (par ...)` could be implemented as a macro at the definition of `PAR64_PA_MASK...`. Then we would have the same code for the 3 flavors (LPAE && ARM32, !LPAE && ARM32, ARM64):

    if (par & PAR_F)
        goto out;
    *pa = PAR_and_VA_to_PA(par,va)